pkg/controller/backup_controller.go

```diff
@@ -260,16 +276,16 @@ func (controller *backupController) processBackup(key string) error {
 		return nil
 	}
 
-	glog.V(4).Infof("running backup for %s", key)
+	logContext.Info("Running backup")
 	// execution & upload of backup
 	if err := controller.runBackup(backup, controller.bucket); err != nil {
-		glog.V(4).Infof("backup %s failed: %v", key, err)
+		logContext.WithField("error", err).Info("backup failed")
 		backup.Status.Phase = api.BackupPhaseFailed
 	}
 
-	glog.V(4).Infof("updating backup %s final status", key)
+	logContext.Info("Updating backup's final status")
 	if _, err = controller.client.Backups(ns).Update(backup); err != nil {
-		glog.V(4).Infof("error updating backup %s final status: %v", key, err)
+		logContext.WithField("error", err).Info("error updating backup's final status")
```

Comment:
what about just

return controller.client.Backups(ns).Update(backup)

You can also use pkg/errors.WithMessage to do this conditionally

return errors.WithMessage(err, "updating backup's final state")